File: InternHub/InternHub.Model/Notification.cs
using InternHub.Model.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InternHub.Model
{
    public class Notification : BaseModel, INotification
    {
        public string ReceiverUserId { get; set; }
        public User ReceiverUser { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
    }
}

File: internhub-frontend/src/models/Notification.js
import { User } from "./User";

export class Notification {
  constructor({ title = "", body = "", user = null }) {
    this.title = title;
    this.body = body;
    this.user = user;
  }

  static fromJson(json) {
    return new Notification({
      title: json["Title"],
      body: json["Body"],
      user: json["User"] ? User.fromJson(json["User"]) : null,
    });
  }
}
